plugin-name/includes/class-plugin-name-deactivator.php

```diff
@@ -22,14 +23,132 @@
 class Plugin_Name_Deactivator {
 
 	/**
-	 * Short Description. (use period)
+	 * The $_REQUEST during plugin activation.
 	 *
-	 * Long Description.
+	 * @since    1.0.0
+	 * @access   private
+	 * @var      array    $request    The $_REQUEST array during plugin activation.
+	 */
+	private static $request = array();
+
+	/**
+	 * The $_REQUEST['plugin'] during plugin activation.
+	 *
+	 * @since    1.0.0
+	 * @access   private
+	 * @var      string    $plugin    The $_REQUEST['plugin'] value during plugin activation.
+	 */
+	private static $plugin  = 'plugin-name/plugin-name.php';
+
+	/**
+	 * The $_REQUEST['action'] during plugin activation.
+	 *
+	 * @since    1.0.0
+	 * @access   private
+	 * @var      array    $action    The $_REQUEST[action] value during plugin activation.
+	 */
+	private static $action  = 'deactivate';
+
+	/**
+	 * Activate the plugin.
+	 *
+	 * Checks if the plugin was (safely) activated.
+	 * Place to add any custom action during plugin activation.
 	 *
 	 * @since    1.0.0
 	 */
 	public static function deactivate() {
 
+		if ( false === self::get_request()
+			|| false === self::validate_request( self::$plugin, self::$action )
+			|| false === self::check_caps()
+			|| ! check_admin_referer( 'deactivate-plugin_' . self::$request['plugin'] )
+		 ) {
+
+			exit;
+
+		}
+
+		/**
+		 * The plugin is now safely activated.
+		 * Perform your activation actions here.
+		 */
+
+	}
+
+	/**
+	 * Get the request.
+	 *
+	 * Gets the $_REQUEST array and checks if necessary keys are set.
+	 * Populates self::request with necessary and sanitized values.
+	 *
+	 * @since    1.0.0
+	 * @return bool|array false or self::$request array.
+	 */
+	private static function get_request() {
+
+		if ( ! empty( $_REQUEST ) ) {
+
+			if ( isset( $_REQUEST['plugin'] )
+				&& isset( $_REQUEST['action'] ) ) {
+
+				self::$request['plugin'] = sanitize_text_field( wp_unslash( $_REQUEST['plugin'] ) );
+				self::$request['action'] = sanitize_text_field( wp_unslash( $_REQUEST['action'] ) );
```

Comment:
`Processing form data without nonce verification`
This must be a false warning, because on my local, where I have WPCS as well, this complaint does not happen. I also doubt it really required to check nonce when we check existence or value of a $_REQUEST?